inject/src/main/java/io/micronaut/context/AbstractBeanDefinition.java

```diff
@@ -128,7 +128,7 @@ protected AbstractBeanDefinition(Class<T> producedType,
                                      boolean requiresReflection,
                                      Argument... arguments) {
 
-        AnnotationMetadata beanAnnotationMetadata = getAnnotationMetadata();
+        getAnnotationMetadata();
```

Comment:
What is the reasoning for this change?

Reply:
`beanAnnotationMetadata ` is not used.
Is the call mandatory?

Reply:
The only thing it will do is initialize the annotation metadata, which should not need done at that point in time. You can try removing the call and see if the build passes

Reply:
Call removed.